### DISCUSSION TITLE
Transaction can be linked to itself

### DISCUSSION THREAD

**Discussion by AlexKalopsia** [firefly-iii/firefly-iii#4330] [General]
Not sure if this is intentional or a defect/bug. I often have transactions such as:

1) Rent February 2020 -> Expense (what I spend for rent)
2) Rent February 2020 <- Income (this is a partial refund from my partner)

Now, I always try and link transactions, but one annoying thing is that if I add a link to 2 and try to look for "Rent February 2020" I find both 1 and 2 in the suggested results. Basically the search field suggests me to link my transition to itself.

My current workaround is that I check the current page url to figure out which transaction I am editing, and then select the other one by exclusion.

My suggestion would be to hide the current transaction from the search results. (This obviously unless I am missing an obvious reason why you'd want to link a transaction to itself)

**Comment by JC5** [comment]
Yes, this is possible in the current layout. Not sure where I did forget to check for that 😅 .

It'll be blocked in the new layout though, since the API doesn't allow it.
